lib/danger/request_source/github.rb

```diff
@@ -153,9 +176,148 @@ def delete_old_comments!(except: nil, danger_id: 'danger')
         end
       end
 
+      def submit_inline_comments!(warnings: [], errors: [], messages: [], markdowns: [], previous_violations: [], danger_id: 'danger')
+        filter = proc do |v|
+          v.file && v.line
+        end
+
+        # Avoid doing any fetchs if there's no inline comments
+        return if (warnings + errors + messages).select(&filter).empty?
+
+        diff_lines = self.pr_diff.lines
+        pr_comments = client.pull_request_comments(ci_source.repo_slug, ci_source.pull_request_id)
+        danger_comments = pr_comments.select { |comment| comment[:body].include?("generated_by_#{danger_id}") }
+
+        submit_inline_comments_for_kind!('warning', warnings, diff_lines, danger_comments, previous_violations[:warning], danger_id: danger_id)
+        submit_inline_comments_for_kind!('no_entry_sign', errors, diff_lines, danger_comments, previous_violations[:error], danger_id: danger_id)
+        submit_inline_comments_for_kind!('book', messages, diff_lines, danger_comments, previous_violations[:message], danger_id: danger_id)
+        submit_inline_comments_for_kind!(nil, markdowns, diff_lines, danger_comments, [], danger_id: danger_id)
+
+        # submit removes from the array all comments that are still in force
+        # so we strike out all remaining ones
+        danger_comments.each do |comment|
+          violation = violations_from_table(comment[:body]).first
+          next if violation.nil?
+
+          # TODO: Check that this violation isn't now out of the diff (case of a force push)
+
+          body = generate_inline_comment_body('white_check_mark', violation, danger_id: danger_id, resolved: true, template: "github")
+          client.update_pull_request_comment(ci_source.repo_slug, comment[:id], body)
+        end
+      end
+
+      def messages_are_equivalent(m1, m2)
+        blob_regexp = %r{blob/[0-9a-z]+/}
+        m1.file == m2.file && m1.line == m2.line &&
+          m1.message.sub(blob_regexp, "") == m2.message.sub(blob_regexp, "")
+      end
+
+      def submit_inline_comments_for_kind!(emoji, messages, diff_lines, danger_comments, previous_violations, danger_id: "danger")
+        head_ref = pr_json[:head][:sha]
+        previous_violations ||= []
+        submit_inline = proc do |m|
+          next false unless m.file && m.line
+
+          position = find_position_in_diff diff_lines, m
+
+          # Keep the change if it's line is not in the diff
+          next false if position.nil?
+
+          # Once we know we're gonna submit it, we format it
+          if emoji.nil?
```

Comment:
any chance you can turn this into a variable? `is_markdown_content = emoji.nil?` maybe? wasn't intuitive at first glance